docs/dev/release-management.md

```diff
@@ -0,0 +1,74 @@
+# Tagging a release
+
+## New major release branch
+
+Pre-requisites for creating a release branch for a new major version:
+
+1. Bump [Go package version](https://github.com/cosmos/ibc-go/blob/main/go.mod#L3).
+2. Change all imports. For example: if the next major version is `v3`, then change all imports starting with `github.com/cosmos/ibc-go/v2` to `github.com/cosmos/ibc-go/v3`).
+
+Once the above pre-requisites are satified:
+
+1. Start on `main`.
+2. Create the release branch (`release/vX.XX.X`). For example: `release/v3.0.x`.
+
+## New minor release branch
+
+1. Start on the latest release branch in the same major release line. For example: the latest release branch in the `v3` release line is `v3.2.x`.
+2. Create branch from the release branch. For example: create branch `release/v3.3.x` from `v3.2.x`.
+
+Post-requisites for both new major and minor release branches:
+
+1. Add branch protection rules to new release branch.
+2. Add backport task to [`mergify.yml`](https://github.com/cosmos/ibc-go/blob/main/.github/mergify.yml).
+3. Create label for backport (e.g.`backport-to-v3.0.x`).
+
+## Point release procedure
+
+In order to alleviate the burden for a single person to have to cherry-pick and handle merge conflicts of all desired backporting PRs to a point release, we instead maintain a living backport branch, where all desired features and bug fixes are merged into as separate PRs.
+
+### Example
+
+Current release is `v1.0.2`. We then maintain a (living) branch `release/v1.0.x`, given `x` as the next patch release number (currently `v1.0.3`) for the `v1.0` release series. As bugs are fixed and PRs are merged into `main`, if a contributor wishes the PR to be released into the `v1.0.x` point release, the contributor must:
+
+1. Add the `backport-to-v1.0x` label to the PR.
+2. Once the PR is merged, the Mergify GitHub application will automatically copy the changes into another branch and open a new PR agains the desired `release/v1.0.x` branch.
+3. Update the PR's description and ensure it contains the following information:
+   - **[Impact]** Explanation of how the bug affects users or developers.
+   - **[Test Case]** section with detailed instructions on how to reproduce the bug.
+   - **[Regression Potential]** section with a discussion how regressions are most likely to manifest, or might
+     manifest even if it's unlikely, as a result of the change. **It is assumed that any backport PR is
+     well-tested before it is merged in and has an overall low risk of regression**. This section should discuss
+     the potential for state breaking changes to occur such as through out-of-gas errors. 
```

Comment:
This is ideal but we don't do this in practice. I'm okay noting that this can informally take place via discussion on the original pr